cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -738,9 +738,7 @@ if(DEFINED ENV{ARROW_ORC_URL})
   set(ORC_SOURCE_URL "$ENV{ARROW_ORC_URL}")
 else()
   set_urls(ORC_SOURCE_URL
-           "https://www.apache.org/dyn/closer.cgi?action=download&filename=/orc/orc-${ARROW_ORC_BUILD_VERSION}/orc-${ARROW_ORC_BUILD_VERSION}.tar.gz"
```

Comment:
We must keep this and use this as the first candidate.
This redirects to downloads.apache.org , dlcdn.apache.org or something for files that exist in https://dist.apache.org/repos/dist/release/orc/ , archive.apache.org otherwise.

BTW, it seems that `closer.cgi` is deprecated. We should use `closer.lua` instead: https://infra.apache.org/release-download-pages.html#download-page